azure-storage-cpp-lite/src/blob/blob_client_wrapper.cpp

```diff
@@ -456,7 +456,14 @@ namespace microsoft_azure {
             }
 
             int result = 0;
-            int block_size = 4*1024*1024;
+
+            //support blobs up to 4.77TB
+            //need to round to the nearest multiple of 4MB for efficiency
+            unsigned long long min_block = fileSize / 50000; 
+            int remainder = min_block % 4*1024*1024;
+            min_block += 4*1024*1024 - remainder;
+            unsigned long long block_size = min_block < MIN_UPLOAD_CHUNK_SIZE ? MIN_UPLOAD_CHUNK_SIZE : min_block;
```

Comment:
Couple things:
- We should check that the file is no greater than 100 * 1024 * 1024 * 50 000 bytes
- We should make sure that the rounding up doesn't go over 100 MB - I don't think it can with the above check, but it won't hurt, just     block_size = block_size > 100MB ? 100MB : block_size
- Very minor - if we first check if (fileSize > (50000 * MIN_UPLOAD_CHUNK_SIZE)), we can skip the math in the majority of cases (any blobs smaller than 800 GB)